chat_shell/chat_shell/agents/graph_builder.py

```
@@ -262,15 +285,33 @@ async def stream_tokens(
 
         event_count = 0
         streamed_content = False  # Track if we've streamed any content
         final_content = ""  # Store final content for non-streaming fallback
 
+        # TTFT tracking variables
+        first_token_received = False
+        llm_request_start_time: float | None = None
+        ttft_ms: float | None = None  # Time to first token in milliseconds
+
+        # Get tracer for LLM request span
+        tracer = otel_trace.get_tracer("chat_shell.agents")
```

Ruff (0.14.11)

Local variable `tracer` is assigned to but never used

Remove assignment to unused variable `tracer`

(F841)